core/options.js

```diff
@@ -127,7 +127,19 @@ Blockly.Options = function(options) {
           Blockly.Colours.hasOwnProperty(colourProperty)) {
         // If a property is in both colours option and Blockly.Colours,
         // set the Blockly.Colours value to the override.
-        Blockly.Colours[colourProperty] = colours[colourProperty];
+        // Ensure to override old Blockly category colours by reference instead
+        // of overriding them, as theya re used by reference elsewhere.
```

Comment:
typo: 'theya re' -> 'they are'

I think this comment is a little confusing as is. I would say it's enough to say something along the lines of "Override Blockly category colors with those provided."